nsxt/resource_nsxt_policy_gateway_flood_protection_profile.go

```diff
@@ -0,0 +1,161 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware/terraform-provider-nsxt/api/infra"
+)
+
+func resourceNsxtPolicyGatewayFloodProtectionProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyGatewayFloodProtectionProfileCreate,
+		Read:   resourceNsxtPolicyGatewayFloodProtectionProfileRead,
+		Update: resourceNsxtPolicyGatewayFloodProtectionProfileUpdate,
+		Delete: resourceNsxtPolicyFloodProtectionProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+		Schema: getGatewayFloodProtectionProfile(),
+	}
+}
+
+func getGatewayFloodProtectionProfile() map[string]*schema.Schema {
+	baseProfile := getFloodProtectionProfile()
+	baseProfile["nat_active_conn_limit"] = &schema.Schema{
+		Type:         schema.TypeInt,
+		Description:  "Maximum limit of active NAT connections",
+		Optional:     true,
+		ValidateFunc: validation.IntBetween(1, 4294967295),
+		Default:      4294967295,
+	}
+	return baseProfile
+}
+
+func resourceNsxtPolicyGatewayFloodProtectionProfilePatch(d *schema.ResourceData, m interface{}, id string) error {
+	connector := getPolicyConnector(m)
+
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+	tags := getPolicyTagsFromSchema(d)
+
+	icmpActiveFlowLimit := int64(d.Get("icmp_active_flow_limit").(int))
+	otherActiveConnLimit := int64(d.Get("other_active_conn_limit").(int))
+	tcpHalfOpenConnLimit := int64(d.Get("tcp_half_open_conn_limit").(int))
+	udpActiveFlowLimit := int64(d.Get("udp_active_flow_limit").(int))
+	natActiveConnLimit := int64(d.Get("nat_active_conn_limit").(int))
+
+	obj := model.GatewayFloodProtectionProfile{
+		DisplayName:  &displayName,
+		Description:  &description,
+		Tags:         tags,
+		ResourceType: model.FloodProtectionProfile_RESOURCE_TYPE_GATEWAYFLOODPROTECTIONPROFILE,
+	}
+
+	if icmpActiveFlowLimit != 0 {
+		obj.IcmpActiveFlowLimit = &icmpActiveFlowLimit
+	}
+	if otherActiveConnLimit != 0 {
+		obj.OtherActiveConnLimit = &otherActiveConnLimit
+	}
+	if tcpHalfOpenConnLimit != 0 {
+		obj.TcpHalfOpenConnLimit = &tcpHalfOpenConnLimit
+	}
+	if udpActiveFlowLimit != 0 {
+		obj.UdpActiveFlowLimit = &udpActiveFlowLimit
+	}
```

Comment:
Minor comment (no need to address): I think it should be possible to have a common function to read/set attributes which are shared between the Gateway and Distributed profiles.